tests/kubernetes/conftest.py

```diff
@@ -0,0 +1,110 @@
+from tests.kubernetes.minikube import *
+from tests.kubernetes.utils import *
+import docker
+import pytest
+
+
+@pytest.fixture(scope="module")
+def local_registry(request):
+    client = docker.from_env(version='auto')
+    final_agent_image_name = request.config.getoption("--k8s-agent-name")
+    final_agent_image_tag = request.config.getoption("--k8s-agent-tag")
+    try:
+        final_image = client.images.get(final_agent_image_name + ":" + final_agent_image_tag)
+    except:
+        try:
+            print("\nAgent image '%s:%s' not found in local registry." % (final_agent_image_name, final_agent_image_tag))
+            print("\nAttempting to pull from remote registry ...")
+            final_image = client.images.pull(final_agent_image_name, tag=final_agent_image_tag)
+        except:
+            final_image = None
+    assert final_image, "agent image '%s:%s' not found!" % (final_agent_image_name, final_agent_image_tag)
+    try:
+        client.containers.get("registry")
+        print("\nRegistry container localhost:5000 already running")
+    except:
+        try:
+            client.containers.run(
+                image='registry:latest',
+                name='registry',
+                detach=True,
+                ports={'5000/tcp': 5000})
+            print("\nStarted registry container localhost:5000")
+        except:
+            pass
+        print("\nWaiting for registry container localhost:5000 to be ready ...")
+        start_time = time.time()
+        while True:
+            assert (time.time() - start_time) < 30, "timed out waiting for registry container to be ready!"
+            try:
+                client.containers.get("registry")
+                time.sleep(2)
+                break
+            except:
+                time.sleep(2)
+    agent_image_name = getattr(request.module, "AGENT_IMAGE_NAME")
+    agent_image_tag = getattr(request.module, "AGENT_IMAGE_TAG")
+    print("\nTagging %s:%s as %s:%s ..." % (final_agent_image_name, final_agent_image_tag, agent_image_name, agent_image_tag))
+    final_image.tag(agent_image_name, tag=agent_image_tag)
+    print("\nPushing %s:%s ..." % (agent_image_name, agent_image_tag))
+    client.images.push(agent_image_name, tag=agent_image_tag)
+    yield
```

Comment:
Not sure how pytest fixtures work but you probably want to do something like this to make sure the registry gets axed if an exception is thrown:
```
try:
    yield
finally:
   client.containers.get("registry").remove(force=True)
```